tests/test_create.py

```diff
@@ -2979,38 +2980,48 @@ def test_multiline_run_command():
     assert not errs_etc
 
 
-def _check_create_xz_env_different_platform(prefix, platform):
-    assert exists(join(prefix, "bin", "xz"))
-    # make sure we read the config from PREFIX/.condarc
-    prefix_condarc = Path(prefix, ".condarc")
-    reset_context([prefix_condarc])
-    config_sources = context.collect_all()
-    assert config_sources[prefix_condarc]["subdir"] == platform
+@pytest.mark.parametrize("style", ["cli", "env"])
+def test_create_env_different_platform(
+    style: Literal["cli", "env"],
+    test_recipes_channel: Path,
+    monkeypatch: MonkeyPatch,
+    tmp_env: TmpEnvFixture,
+    conda_cli: CondaCLIFixture,
+):
+    platform = f"{context.subdir.split('-')[0]}-fake"
 
-    stdout, _, _ = run_command(
-        Commands.INSTALL,
-        prefix,
-        "python",
-        "--dry-run",
-        "--json",
-        use_exception_handler=True,
-    )
-    result = json.loads(stdout)
-    assert result["success"]
-    python = next(pkg for pkg in result["actions"]["LINK"] if pkg["name"] == "python")
-    assert python["platform"] == platform
+    # either set CONDA_SUBDIR or pass --platform
+    if style == "cli":
+        # --platforms has explicit choices, patch to use fake subdir
+        monkeypatch.setattr("conda.base.constants.KNOWN_SUBDIRS", [platform])
 
+        args = [f"--platform={platform}"]
+    else:
+        monkeypatch.setenv("CONDA_SUBDIR", platform)
+        reset_context()
+        assert context.subdir == platform
 
-def test_create_env_different_platform_cli_flag():
-    platform = "linux-64" if on_mac else "osx-64"
-    with make_temp_env("xz", "--platform", platform) as prefix:
-        _check_create_xz_env_different_platform(prefix, platform)
+        args = []
 
+    with tmp_env(*args) as prefix:
+        assert (prefix / ".condarc").read_text() == f"subdir: {platform}\n"
 
-def test_create_env_different_platform_env_var():
-    platform = "linux-64" if on_mac else "osx-64"
-    with env_var("CONDA_SUBDIR", platform), make_temp_env("xz") as prefix:
-        _check_create_xz_env_different_platform(prefix, platform)
+        stdout, stderr, code = conda_cli(
+            "install",
+            f"--prefix={prefix}",
+            "arch-package",
+            "--dry-run",
+            "--json",
+            raises=DryRunExit,
+        )
+        result = json.loads(stdout)
+        assert result["success"]
+        assert any(
+            pkg["name"] == "arch-package" and pkg["platform"] == platform
+            for pkg in result["actions"]["LINK"]
+        )
+        assert not stderr
+        assert code  # DryRunExit
```

Comment:
I thought DryRunExit had exit code 0, since it inherits from `CondaExitZero`:

https://github.com/conda/conda/blob/4064f528bca7281115cbf07eea82527a14d4c821/conda/__init__.py#L123-L124

Reply:
we changed `conda_cli` to return the pytest `ExceptionInfo` object: https://github.com/conda/conda/blob/4064f528bca7281115cbf07eea82527a14d4c821/conda/testing/__init__.py#L193-L195

Reply:
reworked it to:

```python
assert isinstance(excinfo.value, DryRunExit)
```